docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+- **info.yml** - Also known as the manifest file, defines all the different settings that this app requires when it is installed. 
+Settings are often useful if you want reusable apps and you don't want to hard code any values in the app itself.
+- **python/app/dialog.py** - This contains the logic and event callbacks etc. that produce the main app window.
+- **python/app/ui** - This folder contains the automatically generated UI code and resource file. You don't edit this directly, instead, you edit the Qt UI file in the resources folder.
+- **resources/** - In the resources folder, the `dialog.ui` file is a Qt Designer file that you can open up and use to rapidly design and define the look and feel of the app. 
+Once you have made changes you have to execute the `build_resources.sh` script which will convert the UI file to python code and store it as the `/python/app/ui/dialo.py`.
+- **style.qss** - You can define QSS (Qt style sheets) for your UI in this file.
+
+{% include info title="Note" content="An app doesn't need to have a UI however, and the minimum requirements for a valid app are an `app.py` containing an `Application` class and an `info.yml`." %}
+
+### Configuration settings
+Inside the manifest file, there should be a `configuration` section where you can define app settings.
+
+Defining a setting in the manifest file, allows you to configure different setting values for your app in the environment YAML files.
+This is useful if your app needs to behave differently depending on the environment it is in.
+
+For example, you may wish to have a setting that defines a template to use when saving a file. 
+```yaml
+save_template:
+    type: template
+    default_value: "maya_asset_work"
+    description: The template to use when building the path to save the file into
+    allows_empty: False
+```
+Creating a setting for this, means you don't have to hard code the template name in your app code, 
+and [can instead get the value](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=get_setting#sgtk.platform.Application.get_setting) 
+from the settings defined either by default in the `info.yml` or overridden in the environment YAML file settings.
+
+```python
+template = app.get_setting("save_template")
+```
+This means that you could configure your app to use a different template depending on the environment the app is running in.
+
+You can read more on configuration settings [in the reference documentation](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=allows_empty#the-configuration-section).
+
+### Frameworks
+
+If you know that you will be using frameworks, you can add them to the manifest file.
+
+For example, if you plan to use the widget and shotgun utils frameworks for your app, add the following to the `info.yml`:
```

Comment:
"For example, if you plan to use the Qt Widgets and Shotgun Utils frameworks in your app,..."